tests/func/test_api.py

```diff
@@ -101,33 +102,34 @@ def _set_remote_url_and_commit(repo, remote_url):
     repo.scm.commit("modify remote")
 
 
-# FIXME: this test doesn't use scm ;D
-def test_open_scm_controlled(dvc_repo, repo_dir):
-    stage, = dvc_repo.add(repo_dir.FOO)
+def test_open_scm_controlled(tmp_dir, erepo_dir):
+    erepo_dir.scm_gen({"scm_controlled": "file content"}, commit="create file")
 
-    stage_content = open(stage.path, "r").read()
-    with api.open(stage.path) as fd:
-        assert fd.read() == stage_content
+    with api.open("scm_controlled", repo=fspath(erepo_dir)) as fd:
+        assert fd.read() == "file content"
 
 
-# TODO: simplify, we shouldn't need run.
```

Comment:
I don't think there is an easy way to have a non-cached file without `run`, an alternative could be `add`-ing and modifying stage file, but that is artificial, and probably will not require less work than `run`.